transformer_engine/pytorch/ops/fuser.py

```diff
@@ -135,7 +135,11 @@ def forward(
                     requires_grad = any(any(x.requires_grad for x in xs) for xs in extra_inputs)
             for idx in basic_op_idxs:
                 basic_op_ctxs[idx].requires_grad = requires_grad
-            x.requires_grad_(requires_grad=requires_grad)
+            if requires_grad != x.requires_grad:
+                if requires_grad:
+                    x.requires_grad_()
+                else:
+                    x = x.detach()
```

Comment:
This fixes a te.Sequential bug that was exposed by Mcore. When running in eval mode, we want `x.requires_grad=False` so that the op knows that it doesn't need to prepare for that grad. However, PyTorch sometimes complains if you change a tensor's `requires_grad` from `True` to `False` (i.e. when the tensor is not a leaf in the autograd graph). Detaching the tensor works around this case.